httpd/simpleserver.go

```diff
@@ -22,41 +22,40 @@ type SimpleServer interface {
 }
 
 func NewSimpleServer(conf Conf) SimpleServer {
-	simpleServer := simpleServerImpl{
+	return &simpleServerImpl{
 		conf: conf,
+		mux:  http.NewServeMux(),
+		server: &http.Server{
+			Addr: fmt.Sprintf(":%d", conf.Port),
+		},
 	}
-	simpleServer.server = &http.Server{
-		Addr: fmt.Sprintf(":%v", simpleServer.conf.Port),
-	}
-	simpleServer.mux = http.NewServeMux()
-	return simpleServer
 }
 
 type simpleServerImpl struct {
-	conf Conf
+	conf   Conf
 	server *http.Server
-	mux *http.ServeMux
+	mux    *http.ServeMux
 }
 
-func (simpleServer simpleServerImpl) Port() int {
-	return simpleServer.conf.Port
+func (s *simpleServerImpl) Port() int {
```

Comment:
usually receivers are short.

Reply:
Thanks. Is there a reason why you changed them to be pointers? Curious.